helpers/sm_file/file_type.go

```diff
@@ -44,3 +49,19 @@ const (
 	FileTypeCSV            FileType = "csv"
 	FileTypeStringsdict    FileType = "stringsdict"
 )
+
+// ParseType parse string to known file type
+func ParseType[T intStringer](first, last T, typ string) (T, bool) {
```

Comment:
It seems this code will not parse `javaProperties`. This is what user will pass to us because of the [documentation](https://help.smartling.com/hc/en-us/articles/360007894714-Java-Properties).
I sent you [in slack](https://smartling.slack.com/archives/D08VCGF3YE5/p1755278488311149?thread_ts=1755269704.206019&cid=D08VCGF3YE5) a magic that is used in TMS code. It seems you can simply remove `_` before comparing strings.